### ISSUE TITLE
MASGAU stops a multi-file restore if one needs to be elevated

### ISSUE THREAD

**Issue by sanmadjack** [MASGAU/MASGAU#42]
MASGAU should continue restoring files even if one needed elevation

**Comment by sanmadjack** [comment]
Or should it? thoughts?

**Comment by AdmiringWorm** [comment]
I think it should continue, however i also think in single user mode it should ignore file-restores needing elevation.
After all, you already have the all user mode which runs elevated.
But that is of course just my opinion...

**Comment by sanmadjack** [comment]
That's not a good idea. The only way to know if elevation is needed is by trying to restore and getting permission denied errors, which triggers the elevation request, so it can't be 100% accurately predicted without trying. Also if we just skip them, then the user is not aware of these not being restored unless we pop up a dialog, at which point we might as well ask "hey, do you want to elevate?"

As far as all users mode, there are users out there who cannot run as admin. Without elevating, that means they have to find an admin, have them log in to their account (or restart MASGAU as the admin user) and restore the files. Elevation just means they need to get their admin to type in a password and the restore can continue, and it has the security advantage of once the restore is done, the admin privileges are gone. If we only used All Users Mode, then the admin would have to let the user use MASGAU as an admin for as long as MASGAU kept running, which the admin may not like. The elevation is just more convenient for everyone.

**Comment by sanmadjack** [comment]
Obviously this isn't an issue for people who just run in All Users Mode

**Comment by AdmiringWorm** [comment]
hmm, I did not think of that. Yeah you are correct then. but if there are several archives to be restored, and the user has already denied to elevate there would be a bad decision to prompt multiple times.

**Comment by sanmadjack** [comment]
I kind of agree. If something like that goes wrong, things should just stop.

**Comment by AdmiringWorm** [comment]
yeah, or continue while ignoring other restores needing elevation.

**Comment by sanmadjack** [comment]
I'd consider that inconsistent. Better the user is notified, the restore stops, and then the user can decide how to proceed, which would probably be to use All Users Mode
